packages/compartment-mapper/src/node-modules.js

```diff
@@ -309,32 +272,34 @@ const graphPackage = async (
   { packageLocation, packageDescriptor },
   conditions,
   dev,
-  commonDependencyDescriptors,
   languageOptions,
   strict,
-  preferredPackageLogicalPathMap = new Map(),
-  logicalPath = [],
+  {
+    commonDependencyDescriptors = {},
+    preferredPackageLogicalPathMap = new Map(),
+    logicalPath = [],
+    log = console.warn,
+  } = {},
 ) => {
   if (graph[packageLocation] !== undefined) {
     // Returning the promise here would create a causal cycle and stall recursion.
     return undefined;
   }
 
   if (packageDescriptor.name !== name) {
-    console.warn(
-      `Package named ${q(
-        name,
-      )} does not match location ${packageLocation} got (${q(
-        packageDescriptor.name,
-      )})`,
-    );
+    log('Package name does not match location', {
+      name,
+      packageDescriptorName: packageDescriptor.name,
+      packageLocation,
+    });
```

Comment:
I do not mind this warning being silenced by default (noop log). There are legitimate cases.